lib/hanami/middleware/body_parser/parser.rb

```diff
@@ -11,11 +11,15 @@ class BodyParser
       # This is raised when parser fails to parse the body
       #
       # @since x.x.x
-      class BodyParsingError < Hanami::Routing::Parsing::BodyParsingError
+      class BodyParsingError < Hanami::Routing::Error
```

Comment:
@GustavoCaso This error is `Hanami::Middleware::BodyParsingError`, it shouldn't inherit from an error from another namespace like `Hanami::Routing`.

---

We should probably create a `Hanami::Middleware::Error` in a third party lib like `hanami-utils`. So it will be available to all the gems that need to provide a middleware for Hanami.

Reply:
Hi @jodosha great catch, I will add the `Hanami::Middleware::Error` to `hanami-utils` and update the code to use it 😄